### PR TITLE
[TAS-965]🔥 Remove fileSrc from reader query

### PR DESCRIPTION
**Pull request by AuroraHuang22**
Use iscnContentUrls as fileSrc.

### CODE REVIEW

File: src/pages/reader.vue [hunk 1]
```diff
@@ -38,13 +38,8 @@ export default {
...
     fileSrc() {
-      const { src } = this.$route.query;
-      // TODO: check src exists in ISCN
-      // if (src && this.iscnContentUrls.find(url => url === src)) {
-      if (src) {
-        return src;
-      }
-      return this.iscnContentUrls.find(url => url.includes('pdf'));
+      const { format: type } = this.$route.query;
+      return this.iscnContentUrls.find(url => url.includes(type));
```
Comment (williamchong) [inline src/pages/reader.vue:42]: It is possible that one iscn contains more than 1 pdf/epub, maybe use index as well?

Comment (AuroraHuang22) [reply src/pages/reader.vue]: [🐛 Fix invalid file-src](https://github.com/likecoin/liker-land/pull/1525/commits/66850b5abd068b9227c6ac4b9e29c39181b0e427)

File: src/util/nft-book.js [hunk 2]
```diff
@@ -21,3 +22,10 @@ export function getFilenameFromURL(url) {
 export function getDownloadFilenameFromURL(url) {
   return `${getFilenameFromURL(url) || 'content'}`;
 }
+
+export function normalizeURLToHTTP(url) {
+  const [schema, path] = url.split('://');
+  if (schema === 'ar') return `${ARWEAVE_ENDPOINT}/${path}`;
+  if (schema === 'ipfs') return `${IPFS_VIEW_GATEWAY_URL}/${path}`;
+  return url;
+}
```
Comment (nwingt) [inline src/util/nft-book.js:31]: A similar function already exists https://github.com/likecoin/liker-land/blob/5b065d16a06c5639f20b615ae6d3934778bd82a8/src/util/nft.js#L423-L429

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/pages/reader.vue (hunk 1 above)
Comment (williamchong) [reply src/pages/reader.vue]: I mean this won't work if there is like 2 content urls
```
- ar://epub_ep1
- ar://epub_ep2
```
If we only check for `epub`, people can never access the 2nd url @AuroraHuang22

File: src/util/nft-book.js (hunk 2 above)
Comment (AuroraHuang22) [reply src/util/nft-book.js]: [🐛 Fix invalid file-src](https://github.com/likecoin/liker-land/pull/1525/commits/94c268484b7890bd6cf9730678b36360853fe470)

File: src/pages/reader.vue (hunk 1 above)
Comment (AuroraHuang22) [reply src/pages/reader.vue]: [🎨 Add 'index' to query for retrieving specific 'fileSrc'](https://github.com/likecoin/liker-land/pull/1525/commits/b1ed3419a3610947acdb39a388219979b5f0c326)

File: src/components/NFTViewOptionList.vue [hunk 3]
```diff
@@ -82,14 +85,14 @@
       </template>
       <template v-else>
         <ButtonV2
-          v-for="contentUrl in normalizedContentURLs"
+          v-for="(contentUrl, index) in normalizedContentURLs"
           :key="contentUrl"
           class="mt-[12px] w-full"
           preset="outline"
-          :text="getContentUrlButtonText(contentUrl)"
```
Comment (williamchong) [inline src/components/NFTViewOptionList.vue:89]: Should not change this?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/NFTViewOptionList.vue (hunk 3 above)
Comment (AuroraHuang22) [reply src/components/NFTViewOptionList.vue:89]: oops ~ misprint!
[🎨 Add 'index' to query for retrieving specific 'fileSrc'](https://github.com/likecoin/liker-land/pull/1525/commits/9179f8dc3dc27a69ddd42c3e5eb706f4bb2262e5)

**Review by williamchong — APPROVED** [review]

**Pull request merged**